test/org/sagebionetworks/bridge/dynamodb/DynamoScheduledActivityDaoMockTest.java

```diff
@@ -190,7 +338,38 @@ public void testOnlyPersistedActivitiesReturned() {
 
         List<ScheduledActivity> activities = TestUtils.runSchedulerForActivities(context);
         // Only mock the return of one of these activities
-        mockMapperResults(Lists.newArrayList(activities.get(0)));
+        // Mocks loading one of the supplied activities.
+        when(mapper.load(any())).thenAnswer(invocation -> {
+            DynamoScheduledActivity thisSchActivity = invocation.getArgument(0);
+            for (ScheduledActivity schActivity : activities) {
+                if (thisSchActivity.getGuid().equals(schActivity.getGuid()) && thisSchActivity.getHealthCode().equals(schActivity.getHealthCode())) {
+                    return thisSchActivity;
+                }
+            }
+            return null;
+        });
+        // Mocks a query that returns all of the activities.
+        final PaginatedQueryList<DynamoScheduledActivity> queryResults = (PaginatedQueryList<DynamoScheduledActivity>) mock(PaginatedQueryList.class);
+        when(queryResults.iterator()).thenReturn(((List<DynamoScheduledActivity>)(List<?>)activities).iterator());
+        when(queryResults.toArray()).thenReturn(activities.toArray());
+        when(mapper.query((Class<DynamoScheduledActivity>) any(Class.class),
+            (DynamoDBQueryExpression<DynamoScheduledActivity>) any(DynamoDBQueryExpression.class)))
+            .thenReturn(queryResults);
+
+        List<DynamoScheduledActivity> dynamoActivities = dynamoScheduledActivities(activities);
+        
+        QueryResultPage<DynamoScheduledActivity> queryResultPage = (QueryResultPage<DynamoScheduledActivity>)mock(QueryResultPage.class);
+        when(queryResultPage.getResults()).thenReturn(dynamoActivities);
+        when(queryResultPage.getLastEvaluatedKey()).thenReturn(null);
+        
+        when(mapper.queryPage((Class<DynamoScheduledActivity>) any(Class.class),
+            (DynamoDBQueryExpression<DynamoScheduledActivity>) any(DynamoDBQueryExpression.class)))
+            .thenReturn(queryResultPage);
+        
+        // Mock a batch load of the activities
+        Map<String,List<Object>> results = Maps.newHashMap();
+        results.put("some-table-name", new ArrayList<Object>(activities));
+        when(mapper.batchLoad(any(List.class))).thenReturn(results);
```

Comment:
This test doesn't have the Test annotation on it.

Also, is there any particular reason we're inlining mockMapperResults() instead of calling the shared method?

Reply:
Added annotation and fixed test (most of it was superfluous).